core/server/api/posts.js

```diff
@@ -55,7 +55,27 @@ posts = {
          * @returns {Object} options
          */
         function modelQuery(options) {
-            return dataProvider.Post.findPage(options);
+            var requestedColumns = [];
+
+            // 'url' and 'author' depend on other columns. Make sure to load them.
+            if (_.isArray(options.columns)) {
+                requestedColumns = _.clone(options.columns);
+                if (_.includes(options.columns, 'url') || _.includes(options.columns, 'author')) {
+                    options.columns.push('id', 'published_at', 'slug', 'author_id');
```

Comment:
instead of defining a specific solution here, i would suggest making this part of the model logic. something like pseudo:

```
fetchPage
  if (options.columns) {
     // only allow permitted columns
     options.columns = _.intersection(options.columns, this.prototype.permittedAttributes());

     // ensure minimum required columns for further logic
     options.columns union this.defaultColumnsToFetch();
  }

// per model (can be empty)
defaultColumnsToFetch: function()
   return 'id, slug....';
```